builtin/credential/jwt/path_config.go

```diff
@@ -387,6 +384,10 @@ func (b *jwtAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Reque
 		return logical.ErrorResponse("both 'oidc_client_id' and 'oidc_client_secret' must be set for OIDC"), nil
 
 	case config.OIDCDiscoveryURL != "":
+		// trim '.well-known/openid-configuration', due to jwt.NewOIDCDiscoveryKeySet()
+		// assuming the `OIDCDiscoveryURL` doesn't account for its presence.
+		// config is saved to the storage without '.well-known' component
+		config.OIDCDiscoveryURL = strings.TrimSuffix(strings.TrimSuffix(config.OIDCDiscoveryURL, "/"), ".well-known/openid-configuration")
```

Comment:
If it does end up trimming off `.well-known/openid-configuration` here, it will end up with a trailing `/` which otherwise it tries to trim (for whatever reason).  I don't know how important that is, but in any case it's inconsistent.  Probably it's enough to add a `/` before `.well-known`.